packages/overlay/src/integrations/sentry/index.ts

```diff
@@ -103,34 +103,51 @@ export default function sentryIntegration(options?: SentryIntegrationOptions) {
   } satisfies Integration<Envelope>;
 }
 
+/**
+ * Implements parser for
+ * @see https://develop.sentry.dev/sdk/envelopes/#serialization-format
+ * @param rawEvent Envelope data
+ * @returns parsed envelope
+ */
 export function processEnvelope(rawEvent: RawEventContext) {
   const { data } = rawEvent;
-  const [rawHeader, ...rawEntries] = data.trim().split(/\n/gm);
-
-  const header = JSON.parse(rawHeader) as Envelope[0];
-  const items: Envelope[1][] = [];
-  for (let i = 0; i < rawEntries.length; i += 2) {
-    // guard both rawEntries[i] and rawEntries[i + 1] are defined and not empty
-    if (!rawEntries[i] || !rawEntries[i + 1]) {
-      continue;
+  let prevCursor = 0;
+  let cursor = data.indexOf('\n');
+  const envelopeHeader = JSON.parse(data.slice(prevCursor, cursor).toString()) as Envelope[0];
+
+  const items: EnvelopeItem[] = [];
+  while (cursor < data.length - 1) {
+    prevCursor = cursor + 1;
+    cursor = data.indexOf('\n', prevCursor);
+    const itemHeaderLine = data.slice(prevCursor, cursor);
+    const itemHeader = JSON.parse(itemHeaderLine.toString()) as EnvelopeItem[0];
+    prevCursor = cursor + 1;
+    const payloadLength = itemHeader.length;
+    if (payloadLength !== undefined) {
+      cursor += payloadLength + 1;
```

Comment:
`cursor` points to the character before the payload start. `+ payloadLength` points it at the last character of the payload (except for empty, but irrelevant) and `+ 1` points it at the newline after payload contents. This is expected going into the next iteration of the loop.

IIUC, in the other branch `cursor` points at the character **after** the the newline, in which case the next item headers would fail to parse. Could you add a test case that contains an item without length, followed by another item?